src/Microsoft.Extensions.Diagnostics.HealthChecks/IHealthCheckService.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// A service which can be used to check the status of <see cref="IHealthCheck"/> instances registered in the application.
+    /// </summary>
+    public interface IHealthCheckService
+    {
+        /// <summary>
+        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing all the health checks registered in the application.
+        /// </summary>
+        /// <remarks>
+        /// The key maps to the <see cref="IHealthCheck.Name"/> property of the health check, and the value is the <see cref="IHealthCheck"/>
+        /// instance itself.
+        /// </remarks>
+        IReadOnlyDictionary<string, IHealthCheck> Checks { get; }
+
+        /// <summary>
+        /// Runs all the health checks in the application and returns the aggregated status.
+        /// </summary>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> which can be used to cancel the health checks.</param>
+        /// <returns>
+        /// A <see cref="Task{T}"/> which will complete when all the health checks have been run,
+        /// yielding a <see cref="CompositeHealthCheckResult"/> containing the results.
+        /// </returns>
+        Task<CompositeHealthCheckResult> CheckHealthAsync(CancellationToken cancellationToken = default);
```

Comment:
That sounds reasonable. Also makes it a little easier to look at adding things like groups/tags in a later version (though we should consider those before RTM as they may require breaking interface changes)

Reply:
Why pass in the checks themselves rather than an `IEnumerable<string>` of check names? The service already has the check instances. Would you ever provide checks not currently registered with the service?

Reply:
Allowing the caller to provide a list of instances saves dereferencing the dictionary, especially if the caller is managing a grouping of checks. But yeah, it could go either way.